src/shared/user-record.c

```diff
@@ -2775,6 +2775,9 @@ int user_record_match(UserRecord *u, const UserDBMatch *match) {
         if (!match)
                 return true;
 
+        if (!uid_is_valid(u->uid))
```

Comment:
apply this to group_record_match too, with gid_is_valid?

Reply:
Added.